tendrl/node_agent/manager/manager.py

```diff
@@ -239,6 +240,62 @@ def on_pull(self, raw_data):
                     self.etcd_client.write(("nodes/%s/Disks/free/%s") % (
                         raw_data["node_id"], disk), "")
 
+            if "network" in raw_data:
+                LOG.info("on_pull, Updating network")
+                interfaces = raw_data['network']
+                subnet_wise_network_details = {}
+                for ni in interfaces:
+                    interfaces[ni]["node_id"] = raw_data["node_id"]
+                    network = Network(interfaces[ni])
+                    interface_json_str = network.to_json_string()
+                    self.etcd_client.write(
+                        (network.__name__) % (raw_data['node_id'], ni),
+                        interface_json_str
+                    )
+                    subnet = interfaces[ni]["subnet"]
+                    subnet = subnet.replace("/", "_")
+                    if subnet in subnet_wise_network_details:
+                        subnet_wise_network_details[subnet].append(
+                            interface_json_str)
+                    elif subnet != "":
+                        subnet_wise_network_details[subnet] = []
+                        subnet_wise_network_details[subnet].append(
+                            interface_json_str)
+                global_network_details = \
```

Comment:
I dont understand the logic here. Do you want to delete the global networks entry for a specific node here so that you can write the data back again? Somehow this removal and then adding the data back is not very optimal looking to me.

Also I feel you need some comments to explain the logic and philosophy behind the same :)